test/e2e/storage/volume_provisioning.go

```diff
@@ -163,6 +163,80 @@ func testDynamicProvisioning(t storageClassTest, client clientset.Interface, cla
 	return pv
 }
 
+// testGlusterDynamicProvisioning tests the same workflow of testDynamicProvisioning expect it wont attach
+// pv to a pod volume and try write/read. This is due to a limitation in heketi which require raw devices to
```

Comment:
Can you use existing `testDynamicProvisioning` but pass new parameter to control attach?

Reply:
I think it make things complex, or when I first tried tests for gluster dynamic provisioner https://github.com/kubernetes/kubernetes/pull/38048, the outcome of the discussion was to keep it seperately from cloud providers , so new function.